torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -268,7 +268,8 @@ def decompress(fut):
 
         for p, q, tensor in zip(ps, qs, high_rank_tensors):
             torch.matmul(p, q.t(), out=tensor)
-            assert not torch.any(torch.isnan(tensor))
+        if torch.cuda.is_available():
+            torch.cuda.synchronize()
```

Comment:
`torch.cuda.synchronize` will use the current device is none is specified, would it be better to explicitly pass in the device that we save as `device` in this function?

Reply:
Thanks for the advice! Will definitely try it. Let me first change more `wait()` into `value()` in this file first, and check if an explicit synchronization is still necessary. If still needed, then will try to restrict the synchronization to the current device.

Reply:
It turns out even after changing some `wait()` to `value()`, the explicit sync is still required.

Created another PR #49711 to explicitly restrict the sync scope as you suggested.